.github/workflows/schedule.py

```diff
@@ -75,7 +89,102 @@ def _load_issue_template(path: str) -> tuple[dict[str, str], str]:
         return front_matter, f.read()
 
 
+def main(templates: list[tuple[str, datetime]], dry_run: bool) -> None:
+    for template, create_date in templates:
+        cmd_lookup, cmd_create = gh_commands(template, create_date)
+        if dry_run:
+            log.info('Would be running a mock command that creates an issue')
+            print(' '.join(cmd_create))
+        else:
+            create_issue(cmd_create, cmd_lookup)
+
+
+def _create(template: str,
+            start: datetime,
+            period_days: int,
+            at_hour: int,
+            ) -> tuple[str, datetime] | None:
+    """
+    Return a tuple of the template and the datetime.now() instance that matches the
+    configured shcedule for the provided template. This logs the schedule check for
+    each, and it only returns what falls within the schedule window.
+    """
+    log.info('Checking configured schedule time window for %r', template)
+    if 0 == (now - start).days % period_days and now.hour == at_hour:
+        return template, now
+    else:
+        log.info('Current time is outside of the configured schedule window')
+
+
 if __name__ == '__main__':
     logging.basicConfig(level=logging.INFO,
                         format='%(asctime)s %(levelname)+7s %(name)s: %(message)s')
-    create_upgrade_issue()
+
+    args = sys.argv[1:]
+    parser = argparse.ArgumentParser(
+        description='Dryrun to display the command used by GH Actions to create an issue'
+    )
+    parser.add_argument('--dry-run',
+                        metavar='MOCKED_DATE',
+                        type=lambda s: datetime.strptime(s, '%Y-%m-%dT%H:%M'),
+                        help='Dry run based on mock date to verify the template issue configuration')
+    args = parser.parse_args(args)
+
+    now = datetime.now(tz)
+    if args.dry_run:
+        dt = args.dry_run
+        now = now.replace(year=dt.year,
+                          month=dt.month,
+                          day=dt.day,
+                          hour=dt.hour,
+                          minute=dt.minute)
+
+    templates = [
+        # Pick any date for `start` and an issue will be created on that date, as
+        # well as every two weeks before and after, but only in the future. Note
+        # that this doesn't mean that the start date has to lie in the future.
+        # For example …
```

Comment:
Comment needs updating now that it isn't specific to the two weeks schedule for `upgrade.md`.